### PR TITLE
cinnamon: package and add default xapps, upgrade to latest

### LINKED ISSUE #109578: Xapps for nixos
**Project description**
Add the linux mint generic viewer apps (xapps) like xreader, xviewer, xed, xplayer and pix to nixpkgs.

**Metadata**

* website: https://linuxmint-developer-guide.readthedocs.io/en/latest/xapps.html
* xviewer source URL: https://github.com/linuxmint/xviewer
* xreader source URL: https://github.com/linuxmint/xreader
* xed source URL: https://github.com/linuxmint/xed
* xplayer source URL: https://github.com/linuxmint/xplayer
* pix source URL: https://github.com/linuxmint/pix
* platforms: linux 
* license: gpl2

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/109578

Also upgrades everything to 5.2.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/cinnamon/cinnamon-session/default.nix [hunk 1]
```diff
@@ -54,6 +55,7 @@ stdenv.mkDerivation rec {
     xorg.libXcomposite
 
     systemd
+    elogind
```
Comment (jtojnar) [inline pkgs/desktops/cinnamon/cinnamon-session/default.nix:58]: systemd should already provide the same features as elogind.

File: pkgs/desktops/cinnamon/cinnamon-session/default.nix [hunk 2]
```diff
@@ -71,6 +73,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     meson
     ninja
+    cmake
```
Comment (jtojnar) [inline pkgs/desktops/cinnamon/cinnamon-session/default.nix:76]: Is CMake really needed or another instance of https://github.com/jtojnar/nixpkgs-hammering/blob/a24b5537fab53dec6dc638576bff19606ad4a33b/explanations/meson-cmake.md?

File: pkgs/desktops/cinnamon/pix/default.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    webkitgtk
+    libwebp
+    librsvg
+    json-glib
+    clutter
+  ];
+
+  meta = with lib; {
+    description = "A generic image viewer from Linux Mint";
+    homepage = "https://github.com/linuxmint/xviewer";
```
Comment (jtojnar) [inline pkgs/desktops/cinnamon/pix/default.nix:70]: Wrong link?

File: pkgs/desktops/cinnamon/xreader/default.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    xapps
+    cairo
+    libxml2
+    libsecret
+    poppler
+    libspectre
+    libgxps
+    webkitgtk
+    nodePackages.mathjax
+    ninja
```
Comment (jtojnar) [inline pkgs/desktops/cinnamon/xreader/default.nix:66]: Likely should go to `nativeBuildInputs`.

File: pkgs/desktops/cinnamon/cinnamon-session/default.nix (hunk 2 above)
Comment (mkg20001) [reply pkgs/desktops/cinnamon/cinnamon-session/default.nix:76]: it seems like it's not explicitly necessary, so I'll remove that

**Pull request merged**